src/code/code_80140E80.c

```diff
@@ -1,21 +1,359 @@
+#define SCALE_MIN 0.032f
+#define SCALE_UNCHANGED 1.0f //!< also maximum scale
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_8014116C.s")
+// Init
+void func_80140E80(Struct_80140E80* this) {
+    bzero(this, sizeof(Struct_80140E80));
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_801411B4.s")
+// Destroy
+void func_80140EA0(Struct_80140E80* this) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80141200.s")
+// internal, only used in func_80141008
+/**
+ * Draw a bg to the specified color image.
+ *
+ * @param[in,out] gfxP      Pointer to current displaylist
+ * @param[in]     bg        BG object to draw
+ * @param[out]    img       Pointer to beginning of destination color image
+ * @param[in]     width     Output width in pixels
+ * @param[in]     height    Output height in pixels
+ * @param[in]     cycleFlag Which BG mode to use
+ */
+void func_80140EAC(Gfx** gfxP, uObjBg* bg, void* img, s32 width, s32 height, s32 cycleFlag) {
+    Gfx* gfx = *gfxP;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_8014151C.s")
+    gDPPipeSync(gfx++);
+    // Set up color image to draw bg to
+    gDPSetColorImage(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_16b, width, img);
+    gDPSetScissor(gfx++, 0, 0, 0, width, height);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80141678.s")
+    // Draw bg in appropriate type
+    gSPObjRenderMode(gfx++, G_OBJRM_ANTIALIAS | G_OBJRM_BILERP);
+    if (!!(cycleFlag & BG_CYC_COPY) != 0) { //! TODO possibly fake, may be a better way
+        gSPBgRectCopy(gfx++, bg);
+    } else {
+        gSPBgRect1Cyc(gfx++, bg);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80141778.s")
+    gDPPipeSync(gfx++);
+    // Reset the color image and scissor to frame's defaults
+    gDPSetColorImage(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_16b, D_801FBBCC, D_0F000000);
+    gSPDisplayList(gfx++, &D_0E000000.setScissor[0]);
+
+    *gfxP = gfx;
+}
+
+// internal
+/**
+ * Set up a BG from a specified source image and draw it to the specified color image with func_80140EAC(), using the
+ * BG's settings. Use func_8014116C() or func_801411B4()
+ *
+ * @param[out] gfxP     Pointer to current displaylist
+ * @param[in] source    Pointer to beginning of source color image
+ * @param[out] img      Pointer to beginning of destination color image
+ * @param[in] width     Output width in pixels
+ * @param[in] height    Output height in pixels
+ * @param[in] x         Left of drawn output
+ * @param[in] y         Top of drawn output
+ * @param[in] scaleX    Amount to rescale the image, (dsdx). No effect if `cycleFlag` is `BG_CYC_COPY`.
+ * @param[in] scaleY    Amount to rescale the image, (dtdy). No effect if `cycleFlag` is `BG_CYC_COPY`.
+ * @param[in] cycleFlag Which BG mode to use
+ */
+void func_80141008(Gfx** gfxP, void* source, void* img, s32 width, s32 height, f32 x, f32 y, f32 scaleX, f32 scaleY,
+                   s32 cycleFlag) {
```

Comment:
```suggestion
void func_80141008(Gfx** gfxP, void* source, TexturePtr img, s32 width, s32 height, f32 x, f32 y, f32 scaleX, f32 scaleY,
                   s32 cycleFlag) {
```
I think you get the picture